src/Overriders/InMemoryOverrider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\FeatureFlags\Overriders;
+
+use Illuminate\Support\Arr;
+use Worksome\FeatureFlags\Contracts\FeatureFlagEnum;
+use Worksome\FeatureFlags\Contracts\FeatureFlagOverrider;
+
+class InMemoryOverrider implements FeatureFlagOverrider
+{
+
+    /**
+     * @param array<string, bool|null> $overrides
+     * @param bool|null $overrideAll
+     */
+    public function __construct(private array $overrides = [], private bool|null $overrideAll = null)
```

Comment:
@owenvoke I would love your opinion here 🕵️ 

I've set it so this InMemory provider may be instantiated with "default" values coming from the config.
But I am unsure if this makes a lot of sense or not 🤔 
(I confess I was having fun with the manager pattern when implementing it 😄 )

Reply:
Hmm, personally I don't think it makes sense to have it in the config. 🤔 I'd say that for usage in tests, etc. (where this is likely to be used), it should be configured with the new `->set()` or `->setAll()` methods. 🤷🏻

However, I'm not entirely sure, as I'm also not a huge fan of having `->set()` / `->setAll()` on the interface. 🤔

Reply:
Yeah I think I agree about the InMemoery, I would remove the default from config part.


I started adding the `set`and `setAll` methods as it made sense being able to swap the override and not have to change the code.
Like the overrider already knows where to look for values so it kind of makes sense for it to know how to store them as well 🤔 
This way the overrider has full responsibility around it getting and setting those 🤔

Reply:
Yeah. I guess it just seemed a bit odd setting config values from outside of the `ConfigRepository` or the config 🤷🏻 But I'm happy whichever way.